### PR TITLE
Refactor parser

### PR DESCRIPTION
**Pull request by PTLux1517**
Moves NaN checks into the parser and replaces MIME type checks.

### CODE REVIEW

**Review by javitrejo14 — APPROVED** [review]

File: src/Calculations.ts [hunk 1]
```diff
@@ -8,11 +8,10 @@ export function computeAngle(points: Array<Point3D|null>): number|null {
...
         return null;
 
-    const first3HasNullOrNaN = points.slice(0,3)
-        .some(point => point===null||isNaN(point.x)||isNaN(point.y)||isNaN(point.z));
+    const first3HasNull = points.slice(0,3).some(point => point===null);
 
-    if (first3HasNullOrNaN)
-        return null;
+    if (first3HasNull) //null markers are hidden in RenderView (and aren't selectable),
+        return null;   //so this case shouldn't be true except for in testing
```
Comment (vpzomtrrfrt) [inline src/Calculations.ts:14]: not quite true, a selected point might become null when moving through time

File: src/DataTypes.ts [hunk 2]
```diff
@@ -33,7 +33,7 @@ export interface Force {
 
 export interface ForceFrame {
     time: number;
-    forces: Array<Force>; //usually 1, at most 2
+    forces: Array<Force|null>; //usually 1, at most 2
```
Comment (vpzomtrrfrt) [inline src/DataTypes.ts:36]: I was assuming you would just not include the force in the array if it's missing

File: src/Parser.ts [hunk 3]
```diff
@@ -83,34 +86,43 @@ export async function parseForceFileData(file: File): Promise<ForceFileData> {
...
+                z: parseFloat(tsvTable[row][force1CompCol+2])
+            },
+            torque: parseFloat(tsvTable[row][force1TrqCol])
+        };
+        if (isNaN(force1.position.x) || isNaN(force1.position.y) || isNaN(force1.position.z)
+           || (force1.position.x===0 && force1.position.y===0 && force1.position.z===0)
+        ) {
+            force1 = null;
+        }
+        let force2: Force|null = {
```
Comment (vpzomtrrfrt) [inline src/Parser.ts:107]: would be nicer to deduplicate this

File: src/Calculations.ts (hunk 1 above)
Comment (PTLux1517) [reply src/Calculations.ts]: True. I forgot the user might take an angle measurement while paused, leave the markers selected, and drag the timeline thumb to another moment. I was picturing taking measurements while the scene is static. Fixed

File: src/Parser.ts (hunk 3 above)
Comment (PTLux1517) [reply src/Parser.ts]: Fixed. I usually prefer conciseness too, but couldn't immediately think of a readable solution with the nature of the columns being a bit scattered in the data.

**Review by vpzomtrrfrt — APPROVED** [review]

**Pull request merged**